lib/http/options.rb

```diff
@@ -50,7 +50,8 @@ def initialize(options = {})
         :keep_alive_timeout => 5,
         :headers            => {},
         :cookies            => {},
-        :encoding           => nil
+        :encoding           => nil,
+        :logger             => HTTP::Logger.new
```

Comment:
@tarcieri @ixti 
`HTTP::Logger` implicitly works like a null object. We can call `HTTP::Logger#log` even if its `logger` attribute is `nil`.
https://github.com/httprb/http/pull/348/files/9e0569ead67133f171b33582e515bade4185b970#diff-a32552e6c7aef4937f6b3148e3515abdR11

As an alternative we can use `nil` as default HTTP logger and to deal with this `nil` when `HTTP::Logger#log` is called in `HTTP::Client` class. 😉